libraries/AP_Math/matrix3.cpp

```diff
@@ -228,6 +228,42 @@ T Matrix3<T>::det() const
 }
 
 template <typename T>
+Matrix3<T> Matrix3<T>::inverse(bool& success) const
```

Comment:
I'm going to try @tridge's alternative, but I'd still like to give this implementation a try. After the call, I thought of some ways to potentially reduce the memory usage here significantly.